### PR TITLE
refactor(api): consolidate comunications projects

### PR DESCRIPTION
**Pull request by losolio**

### CODE REVIEW

File: apps/api/tests/Eventuras.TestAbstractions/DisposableEntity.cs [hunk 1]
```diff
@@ -12,75 +13,133 @@ namespace Eventuras.TestAbstractions;
...
+            }
+
+            _disposed = true;
+        }
+    }
+
+    protected virtual async ValueTask DisposeAsyncCore()
+    {
+        // Async dispose managed state (managed objects) if needed.
+        foreach (var disposable in _disposables.OfType<IAsyncDisposable>())
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/tests/Eventuras.TestAbstractions/DisposableEntity.cs:76]: Resolved in https://github.com/losol/eventuras/pull/810

File: apps/api/tests/Eventuras.TestAbstractions/DisposableEntity.cs [hunk 2]
```diff
@@ -12,75 +13,133 @@ namespace Eventuras.TestAbstractions;
...
         {
-            disposable.Dispose();
+            await disposable.DisposeAsync();
         }
 
         try
         {
             _context.Remove(Entity);
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/tests/Eventuras.TestAbstractions/DisposableEntity.cs:83]: Resolved as _context is disposed on tests teardown and entities are disposed inside tests.

File: apps/api/src/Eventuras.WebApi/Extensions/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -164,7 +167,8 @@ public static void AddApplicationHealthChecks(this IServiceCollection services,
             .AddHealthChecksUI(settings =>
             {
                 settings
-                    .AddHealthCheckEndpoint(Constants.HealthChecks.HealthCheckName, $"{baseUri}{Constants.HealthChecks.HealthCheckUri}");
+                    .AddHealthCheckEndpoint(Constants.HealthChecks.HealthCheckName,
+                        $"{baseUri}{Constants.HealthChecks.HealthCheckUri}");
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/src/Eventuras.WebApi/Extensions/ServiceCollectionExtensions.cs:171]: Resolved in https://github.com/losol/eventuras/pull/810

File: apps/api/libs/Losol.Communication.Sms/Twilio/TwilioSmsSender.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    private readonly IOptions<TwilioOptions> _options;
+    private readonly IHealthCheckStorage _healthCheckStorage;
+    private readonly ILogger<TwilioSmsSender> _logger;
+
+    public TwilioSmsSender(
+        IOptions<TwilioOptions> options,
+        IHealthCheckStorage healthCheckStorage,
+        ILogger<TwilioSmsSender> logger)
+    {
+        _options = options ?? throw new ArgumentNullException(nameof(options));
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Sms/Twilio/TwilioSmsSender.cs:26]: Resolved in https://github.com/losol/eventuras/pull/810

File: apps/api/libs/Losol.Communication.Sms/Twilio/TwilioSmsSender.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        if (string.IsNullOrWhiteSpace(body))
+        {
+            throw new ArgumentException(nameof(body));
+        }
+
+        try
+        {
+            var messageResource = await MessageResource.CreateAsync(
+                new PhoneNumber(to),
+                @from: new PhoneNumber(_options.Value.From),
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Sms/Twilio/TwilioSmsSender.cs:49]: Resolved in https://github.com/losol/eventuras/pull/810

File: apps/api/libs/Losol.Communication.HealthCheck/Abstractions/HealthCheckStatus.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
+using System;
+
+namespace Losol.Communication.HealthCheck.Abstractions;
+
+public class HealthCheckStatus
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.HealthCheck/Abstractions/HealthCheckStatus.cs:5]: Ignored.

File: apps/api/libs/Losol.Communication.HealthCheck/Abstractions/AbstractPeriodicHealthCheck.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    protected abstract TimeSpan CheckPeriod { get; }
+
+
+    private readonly IHealthCheckStorage _healthCheckStorage;
+    private readonly IHealthCheckService _healthCheckService;
+
+    protected AbstractPeriodicHealthCheck(IHealthCheckStorage healthCheckStorage, IHealthCheckService healthCheckService)
+    {
+        _healthCheckStorage = healthCheckStorage ?? throw new ArgumentNullException(nameof(healthCheckStorage));
+        _healthCheckService = healthCheckService ?? throw new ArgumentNullException(nameof(healthCheckService));
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.HealthCheck/Abstractions/AbstractPeriodicHealthCheck.cs:26]: Ignored

File: apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        _options = options;
+        if (!Directory.Exists(options.Value.FilePath))
+        {
+            Directory.CreateDirectory(options.Value.FilePath);
+        }
+    }
+
+    protected override async Task SendEmailInternalAsync(EmailModel emailModel)
+    {
+        emailModel.Validate();
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs:31]: Resolved in #810

File: apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+{
+    /// <summary>
+    /// Creates a URL And SEO friendly slug
+    /// Copyright (c) Johan Boström. All rights reserved.
+    /// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+    /// </summary>
+    /// <param name="text">Text to slugify</param>
+    /// <param name="maxLength">Max length of slug</param>
+    /// <returns>URL and SEO friendly string</returns>
+    internal static string GenerateSlug(this string text, int maxLength = 0)
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs:52]: @losolio Need your input

[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs:52]: this is mainly used for filenames i think, and will keep this as is!

File: apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+
+    protected override async Task SendEmailInternalAsync(EmailModel emailModel)
+    {
+        emailModel.Validate();
+
+        // filename: {datetime}-{email}-{subject}.html
+        var filename = $"{DateTime.Now:yyyyMMdd-HHmmss}-{emailModel.Recipients.First().Email.GenerateSlug()}-{emailModel.Subject.GenerateSlug()}.html";
+
+        // Write the message to the file
+        await using var outputFile = new StreamWriter(Path.Combine(_options.Value.FilePath, filename));
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Email/File/FileEmailWriter.cs:37]: Resolved in #810

File: apps/api/libs/Losol.Communication.Email/SendGrid/SendGridEmailSender.cs [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+                Type = a.ContentType,
+                Disposition = a.ContentDisposition,
+                ContentId = a.ContentId
+            }));
+        }
+
+        var client = new SendGridClient(_config.Key);
+        var response = await client.SendEmailAsync(msg);
+        if (response.StatusCode != HttpStatusCode.OK &&
+            response.StatusCode != HttpStatusCode.Accepted)
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Email/SendGrid/SendGridEmailSender.cs:60]: Resolved in #810

File: apps/api/libs/Losol.Communication.Email/Smtp/SmtpEmailSender.cs [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+                await emailClient.AuthenticateAsync(_smtpConfig.Username, _smtpConfig.Password);
+            }
+            await emailClient.SendAsync(mimeMessage);
+            await emailClient.DisconnectAsync(true);
+
+            _logger.LogInformation("*** END SEND EMAIL ***");
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex.Message);
```
[the remark this answers is not in the document]
Comment (yura-b) [reply apps/api/libs/Losol.Communication.Email/Smtp/SmtpEmailSender.cs:100]: Resolved in #810

**Pull request merged**